pytorch_lightning/trainer/train_loop_mixin.py

```diff
@@ -278,10 +278,12 @@ def training_forward(self, batch, batch_nb, opt_idx, hiddens):
         if self.truncated_bptt_steps is not None:
             args.append(hiddens)
 
-        if self.use_ddp or self.use_ddp2:
+        if self.use_ddp:
             output = self.model(*args)
-        elif self.use_dp:
+        elif self.use_ddp2 or self.use_dp:
+            # in dp, allow model to use training_step and training_end
             output = self.model(*args)
```

Comment:
Why break this up into two `if` statements if we're doing the same thing in both cases?